### ISSUE TITLE
No ability to pass time ranges

### ISSUE THREAD

**Issue by JasonKeirstead** [Soltra/cti-stats#3]
After hacking around issue #2 , I started seeing this error

```
<taxii_11:Status_Message xmlns:taxii_11="http://taxii.mitre.org/messages/taxii_xml_binding-1.1" status_type="BAD_MESSAGE" in_response_to="24164">
  <taxii_11:Message>Too Large Date Interval: 16730</taxii_11:Message>
</taxii_11:Status_Message>
```

Our TAXII server only allows you to poll a time range of a couple of days, so to work around this I had to edit utils.py and manually change "epoch_start()".

**Comment by treyka** [comment]
@JasonKeirstead Thanks for the issue, I'll get on this pronto! ^_^

**Comment by JasonKeirstead** [comment]
HI Trey, that actually still doesn't work for my use case as our server does not allow a poll for more than 48 hours .

**Comment by treyka** [comment]
@JasonKeirstead Perhaps I wasn't clear. Based on your feedback, cti-stats no longer attempts an all-time poll. It divides the total polling time into *12 hour blocks* so if your server doesn't permit a poll across > 48 hours, 12 < 48 so it *should* work.

Did you actually test or just go by my comment?

**Comment by JasonKeirstead** [comment]
I tried testing but currently it's just hanging forever at 0%... trying to debug it.

**Comment by JasonKeirstead** [comment]
```File "./cti-stats", line 77, in <module>
    main()
  File "./cti-stats", line 70, in main
    args['--quiet'])
  File "/home/jasonk/cti-stats/lib/cti.py", line 135, in taxii_poll
    poll_request.message_id)
  File "/home/jasonk/cti-stats/python_env/local/lib/python2.7/site-packages/libtaxii/__init__.py", line 50, in get_message_from_http_response
    return get_message_from_urllib_addinfourl(http_response, in_response_to)
  File "/home/jasonk/cti-stats/python_env/local/lib/python2.7/site-packages/libtaxii/__init__.py", line 78, in get_message_from_urllib_addinfourl
    response_message = http_response.read()
  File "/usr/lib/python2.7/socket.py", line 351, in read
    data = self._sock.recv(rbufsize)
  File "/usr/lib/python2.7/httplib.py", line 549, in read
    return self._read_chunked(amt)
  File "/usr/lib/python2.7/httplib.py", line 609, in _read_chunked
    value.append(self._safe_read(amt))
  File "/usr/lib/python2.7/httplib.py", line 664, in _safe_read
    chunk = self.fp.read(min(amt, MAXAMOUNT))
  File "/usr/lib/python2.7/socket.py", line 380, in read
    data = self._sock.recv(left)
  File "/usr/lib/python2.7/ssl.py", line 341, in recv
    return self.read(buflen)
  File "/usr/lib/python2.7/ssl.py", line 260, in read
    return self._sslobj.read(len)
KeyboardInterrupt
```

**Comment by treyka** [comment]
Did you update your pip virtual environment? Because there's one new dependency.

**Comment by treyka** [comment]
@JasonKeirstead Also, could you post the exact calling arguments (sensitive info sanitized, obviously)?

**Comment by JasonKeirstead** [comment]
Yes I did.

I am unsure how to debug this, it just appears to be hanging.

The outbound request looks sane

<taxii_11:Poll_Request xmlns:taxii="http://taxii.mitre.org/messages/taxii_xml_binding-1" xmlns:taxii_11="http://taxii.mitre.org/messages/taxii_xml_binding-1.1" xmlns:tdq="http://taxii.mitre.org/query/taxii_default_query-1" message_id="28488" collection_name="xfe.default"><taxii_11:Exclusive_Begin_Timestamp>2015-10-22T13:35:16+00:00</taxii_11:Exclusive_Begin_Timestamp><taxii_11:Inclusive_End_Timestamp>2015-10-23T01:35:16+00:00</taxii_11:Inclusive_End_Timestamp><taxii_11:Poll_Parameters allow_asynch="false"><taxii_11:Response_Type>FULL</taxii_11:Response_Type><taxii_11:Content_Binding binding_id="urn:stix.mitre.org:xml:1.1"/></taxii_11:Poll_Parameters></taxii_11:Poll_Request>

**Comment by JasonKeirstead** [comment]
<taxii_11:Poll_Request xmlns:taxii="http://taxii.mitre.org/messages/taxii_xml_binding-1" xmlns:taxii_11="http://taxii.mitre.org/messages/taxii_xml_binding-1.1" xmlns:tdq="http://taxii.mitre.org/query/taxii_default_query-1" message_id="28488" collection_name="xfe.default"><taxii_11:Exclusive_Begin_Timestamp>2015-10-22T13:35:16+00:00</taxii_11:Exclusive_Begin_Timestamp><taxii_11:Inclusive_End_Timestamp>2015-10-23T01:35:16+00:00</taxii_11:Inclusive_End_Timestamp><taxii_11:Poll_Parameters allow_asynch="false"><taxii_11:Response_Type>FULL</taxii_11:Response_Type><taxii_11:Content_Binding binding_id="urn:stix.mitre.org:xml:1.1"/></taxii_11:Poll_Parameters></taxii_11:Poll_Request>

**Comment by treyka** [comment]
My long-running process against HaT just stacktraced on some bogus input to I changed a try/except block in 5b66940d35bca59b3acfb2ea8692a3a294f4f253 in an effort to handle malformed stix/cybox more gracefully. Mind trying again?

**Comment by JasonKeirstead** [comment]
OK I know what is going on here, it is just taking forever and a day at 0% due to all the polls...

**Comment by treyka** [comment]
@JasonKeirstead Btw, I would like you to know *just how much* I appreciate your help on this! CTI interoperability is a known issue (otherwise we wouldn't have just created a new OASIS CTI subcommittee on it). I knew from the outset there were likely going to be issues with other implementations (ie, not Soltra Edge). Really, dude, thanks a mil for the assist here!

**Comment by treyka** [comment]
Fixed in 0b19bd292dba856709d33040427cef43d5d19f23; thanks for the pull-request, @JasonKeirstead!
